odl/space/npy_tensors.py

```diff
@@ -0,0 +1,2336 @@
+        >>> x = space.one()
+        >>> x
+        rn((2, 3)).element(
+            [[1.0, 1.0, 1.0],
+             [1.0, 1.0, 1.0]]
+        )
+        """
+        return self.element(np.ones(self.shape, dtype=self.dtype,
+                                    order=self.new_elem_order))
+
+    @staticmethod
+    def available_dtypes():
+        """Return the tuple of data types available in this implementation.
+
+        Notes
+        -----
+        This is all dtypes available in Numpy. See `numpy.sctypes`
+        for more information.
+
+        The available dtypes may depend on the specific system used.
+        """
+        all_dtypes = []
+        for lst in np.sctypes.values():
+            all_dtypes.extend(set(lst))
+        dtypes = [np.dtype(dtype) for dtype in all_dtypes]
+        dtypes.remove(np.dtype('void'))
+        return tuple(dtypes)
+
+    @staticmethod
+    def default_dtype(field=None):
+        """Return the default data type of this class for a given field.
+
+        Parameters
+        ----------
+        field : `Field`, optional
+            Set of numbers to be represented by a data type.
+            Currently supported : `RealNumbers`, `ComplexNumbers`
+            The default ``None`` means `RealNumbers`
+
+        Returns
+        -------
+        dtype : `numpy.dtype`
+            Numpy data type specifier. The returned defaults are:
+
+                ``RealNumbers()`` : ``np.dtype('float64')``
+
+                ``ComplexNumbers()`` : ``np.dtype('complex128')``
+        """
+        if field is None or field == RealNumbers():
+            return np.dtype('float64')
+        elif field == ComplexNumbers():
+            return np.dtype('complex128')
+        else:
+            raise ValueError('no default data type defined for field {}'
+                             ''.format(field))
+
+    def _lincomb(self, a, x1, b, x2, out):
+        """Implement the linear combination of ``x1`` and ``x2``.
+
+        Compute ``out = a*x1 + b*x2`` using optimized
+        BLAS routines if possible.
+
+        This function is part of the subclassing API. Do not
+        call it directly.
+
+        Parameters
+        ----------
+        a, b : `TensorSpace.field` element
+            Scalars to multiply ``x1`` and ``x2`` with.
+        x1, x2 : `NumpyTensor`
+            Summands in the linear combination.
+        out : `NumpyTensor`
+            Tensor to which the result is written.
+
+        Examples
+        --------
+        >>> space = odl.cn((2, 3))
+        >>> x = space.element([[0, 1j, 1],
+        ...                    [0, 1, 0]])
+        >>> y = space.element([[0, 0, 1j],
+        ...                    [-1j, 0, 2j]])
+        >>> out = space.element()
+        >>> space.lincomb(1j, x, 1, y, out)  # out is returned
+        cn((2, 3)).element(
+        [[0j, (-1+0j), 2j],
+         [-1j, 1j, 2j]]
+        )
+        >>> out
+        cn((2, 3)).element(
+            [[0j, (-1+0j), 2j],
```

Comment:
This is the correct one according to the implementation. That's up to debate. Just weird that the other one is accepted, too. Probably due to the `NORMALIZE_WHITESPACE` option of doctest.